routes/api/v2.js

```diff
@@ -0,0 +1,310 @@
+ * by a person on github. This will cache the results to make simultaneous
+ * connections easier and less demanding on the github API.
+ * 
+ * @param {*} user 
+ */
+const queryFriends = async user => {
+    const cacheHit = cache.get("/friends/" + user);
+    if (cacheHit){
+        console.log("Friends cache hit");
+        return cacheHit;
+    }
+
+    try {
+        const followers = await fetchAllWithPagination(API_USER_PATH + user + API_FOLLOWERS, 1, []);
+        const following = await fetchAllWithPagination(API_USER_PATH + user + API_FOLLOWING, 1, []);
+        const fList = minimizeFriends(following.concat(followers));
+        cache.put(`/friends/${user}`, fList);
+        return fList;
+    } catch (error) {
+        console.log("API Error", err); 
+    }
+}
+
+
+/**
+ * 
+ * Fetches all of the members of an organization from the
+ * API or cache
+ *
+ * /orgs/RITlug/members?page=1
+ *
+ * @param {*} orgName 
+ */
+const getOrganizationMembers = async orgName => {
+    const cacheHit = cache.get("/org/users/" + orgName);
+    if (cacheHit){
+        console.log("Org members cache hit");
+        return cacheHit;
+    }
+
+    try {
+        const members = await fetchAllWithPagination(API_ORGS_PATH + orgName + "/members", 1, []);
+        const membersMin = minimizeFriends(members);
+        cache.put("/org/users/" + orgName, membersMin);
+        return membersMin;
+    } catch (error) {
+        console.log(error);
+    }
+}
+
+
+/**
+ * Minimizes the JSON for a list of repositories
+ * 
+ * @param {*} repositories 
+ */
+const minimizeRepositories = repositories => {
+    let rList = [];
+    repositories.forEach(repo => {
+        rList.push(copyWithProperties(["name", "created_at", "homepage", 
+            "description", "language", "forks", "watchers",
+            "open_issues_count", "license", "html_url"],
+            repo));
+    })
+
+    return rList;
+}
+
+
+/**
+ * Fetches all repositories from the API
+ * 
+ * @param {*} user name of org/user
+ * @param {*} orgsOrUsers  either /users/ or /orgs/
+ */
+const queryRepositories = async (user, orgsOrUsers) => {
+    const cacheHit = cache.get(user + REPOS_PATH);
+    if (cacheHit) {
+        console.log("Repositories cache hit");
+        return cacheHit;
+    }
+
+    try {
+        const repos = await fetchAllWithPagination(orgsOrUsers + user + REPOS_PATH, 1, []);
+        const minRepos = minimizeRepositories(repos);
+        cache.put(`${user}${REPOS_PATH}`, minRepos);
+        return minRepos;
+    } catch (error) {
+        console.log(error)
+        console.log("bad things went down");
+    }
+}
+
+
+/**
+ * /users/name/following/followers
+ */
+routes.get("/friends/:name", async (req, res)=> {
+    try {
+        const query = await queryFriends(req.params.name);
+        res.json(query);
+    } catch (error) {
+        res.status(500).json({error: 'API error fetching friends'});
+    }
+});
+
+
```

Comment:
hmm, we should make some API docs somewhere

Reply:
I can work on starting that effort today.

Reply:
https://github.com/jrtechs/github-graphs/issues/29